### DISCUSSION TITLE
How can I toggle flat view using a single key?

### DISCUSSION THREAD

**Discussion by TobisLee** [sxyazi/yazi#1421] [1. Q&A] [answered]
When I use `search fd` to get flat view, I notice that I need to press an extra `Enter` key. To avoid this, I searched the issues and docs, and tried the following methods:
1. Add the following code in `keymap.toml`
```toml
[[manager.prepend_keymap]]
on   = ["T"]
run  = ["search fd", "close --submit"]
desc = "Toggle flat view"
```
With this setting, when I press `T`, yazi quits.
2. Add a plugin
When I searched the issues, I found [this](https://github.com/sxyazi/yazi/discussions/1221#discussioncomment-9926528). I thought it could solve my problem, so I wrote a plugin.
```lua
local function entry(st)
	if st.flatview then
		ya.manager_emit("escape", {})
		st.flatview = false
	else
		ya.manager_emit("search", { "fd" })
                ya.input_emit("close", { submit = true })
		-- ya.input_emit("close_input", { submit = true })
		st.flatview = true
	end
end

return {
	entry = entry,
}
```
However, I still need to press the `Enter` key. Is there something I missed? How can I toggle flat view using a single key?

**Comment by AnirudhG07** [comment]
Have your tried [smart-filter](https://github.com/yazi-rs/plugins/tree/main/smart-filter.yazi)? It automatically enters and reaches your files without pressing enter.

**Reply by TobisLee** [reply]
Thanks for the reply. I checked out the `smart-filter` code, but it doesn't seem to solve my problem because `smart-filter` uses its own input prompt. In my case, the input prompt is generated by `search fd`.

**Comment by sxyazi** [comment]
I can add a new command for it, would you like to file an issue?

**Reply by TobisLee** [reply]
Of course. See #1425

**Reply by TobisLee** [reply, marked as answer]
#1431 really solves my problem. Thanks for the amazing work!
